### DISCUSSION TITLE
Allow change the driver connection

### DISCUSSION THREAD

**Discussion by ceinmart** [dbeaver/dbeaver#36370] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
We have a lot of connections for multiple database.
Few of them, for particular reasons sometimes need to run over an specific driver version. 
Sometimes for version limitation, sometimes for test propose.
It's impossible to change the driver from an connection. 
If I have on my manager 3 drivers configurations for the same vendor (Informix for example) , where eachone is set to a certain version. 
When I need to change a specific connect to another version I need to create a new connection, which is annoying because our connections are quite personalized (driver parameters and dbeaver parameter) and I lost all historical references of the scripts with that connection. 

**Describe the solution you'd like**
Option to change the origin driver without the need to create a new one and keeping the common parameters like host, server, port, password, dbeaver personalization.

**Describe alternatives you've considered**
I don't see any alternatives.
The current available solution, is changing the version of the driver used for that connection, this will affect other connections and that isn't the intention.

**Comment by E1izabeth** [comment]
Thank you for sharing your use case.

**Comment by arhayka** [comment]
We discussed this as a team and concluded that it’s not something we need to implement in the near future. It seems too complicated for users, and copying the driver already works as a workaround.

**Comment by ceinmart** [comment]
Hi @arhayka  , 
Can you please explain how to copy a connection with a different driver while keeping all other connection settings?
Because as far I know and see the option on dbeaver, there is no way to "copying the driver" as workaround. 

Not sure if I was clear on my request. 
Below I put a screenshot of SquirrelSQL (another isql tool, you probably know about it). 
As example on SquirrelSQL, is very simple to change the driver without change others connections parameters.

Here is a connection configuration on SquirrelSQL
![image](https://github.com/user-attachments/assets/fb5772e8-d98d-4819-ae0b-5c58b37bc499)

On this combo , I can change the driver utilized , keeping all other configuration and parametrization of this specific connection.
This appear to be impossible to do on dbeaver.
![image](https://github.com/user-attachments/assets/74af0d30-e1cb-4c0b-820e-53b971e23319)

**Comment by JTMosaic** [comment, marked as answer]
I believe this functionality is already a part of DBeaver. It just takes a little bit of preparation.

Open the Driver Manager from the Database menu
Select the driver you want to use a different version of and click the Copy button
In the properties window that opens, go to the Libraries tab and download the version you wish to use
On the Settings tab, change the Driver Name to indicate the version

You should then be able to change the driver for one or many connections to the copy of the driver without losing any connection information. I do this with MariaDB and MySQL drivers

Make sure the project in which the connection you want to change is selected as active.
Select the connection and then click on the Database menu
Select the last option "Change Connection(s) Driver"

Note this only works if the connection is in the current active project

Does that achieve the desired result?

**Comment by ceinmart** [comment]
> Make sure the project in which the connection you want to change is selected as active.
> Select the connection and then click on the Database menu
> Select the last option "Change Connection(s) Driver"

@JTMosaic  , Thank you!
That is the part I'm missing, I never noticed this menu option. 

Consider this problem solved!
Thank you again.
